src/role_scout.js

```diff
@@ -13,153 +13,219 @@ roles.scout.settings = {
+  config.visualizer.enabled
+  && config.visualizer.showPathSearches
+  && visualizer.showSearch(search);
+
+  if (search.path.length === 0 || (creep.inBase() && creep.room.memory.misplacedSpawn)) {
+    if (creep.isStuck() && roles.scout.onBorder(creep)) {
+      creep.say('imstuck at the border', true);
+      if (config.scout.scoutSkipWhenStuck) {
+        creep.say('skipping', true);
+        creep.memory.scoutSkip = true;
+        delete creep.memory.last; // Delete to reset stuckness.
+      }
     }
 
-    if (search.path.length === 0 || (creep.inBase() && creep.room.memory.misplacedSpawn)) {
-      creep.say('hello', true);
-      //       creep.log(creep.pos + ' ' + targetPosObject + ' ' + JSON.stringify(search));
-      if (creep.isStuck() && onBorder(creep)) {
-        creep.say('imstuck at the border', true);
-        if (config.room.scoutSkipWhenStuck) {
-          creep.say('skipping', true);
-          creep.memory.scoutSkip = true;
-          delete creep.memory.last; // Delete to reset stuckness.
-        }
+    let returnCode = creep.moveTo(creep.targetPosObject, {
+      ignoreCreeps: true,
+      costCallback: creep.room.getCostMatrixCallback()
+    });
+
+    return true;
+  }
+  creep.say('Hi -> ' + creep.memory.search.target, true);
+  let returnCode = creep.move(creep.pos.getDirectionTo(search.path[0]));
+}
+
+roles.scout.breadthFirstSearch = function(creep) {
+  roles.scout.initSearch(creep);
+
+  (creep.memory.scoutSkip || creep.room.name === creep.memory.search.target)
+  && roles.scout.switchTarget(creep);
+
+  if (!creep.memory.search.target) {
+    creep.log('Suiciding: ' + JSON.stringify(creep.memory.search));
+    creep.suicide();
+    return true;
+  }
+
+  creep.targetPosObject = new RoomPosition(25, 25, creep.memory.search.target);
+
+  if (creep.room.getEnemys().length) {
+    creep.say('Afraid', true);
+    creep.memory.scoutSkip = true;
+    return false;
+  }
+  if (creep.memory.last && creep.memory.last.pos3 && creep.room.name !== creep.memory.last.pos3.roomName) {
+    let roomMem = Memory.rooms[creep.room.name];
+    if (!roomMem) {
+      Memory.rooms[creep.room.name] = {};
+      Memory.rooms[creep.room.name].scoutSeen = creep.name;
+    } else {
+      let youngerCreepHere = Memory.rooms[creep.room.name].scoutSeen;
+      if (!youngerCreepHere || !Game.creeps[youngerCreepHere] || Game.creeps[youngerCreepHere].ticksToLive < creep.ticksToLive) {
+        Memory.rooms[creep.room.name].scoutSeen = creep.name;
       }
-      //if (search.path.length > 0) {
-      //creep.move(creep.pos.getDirectionTo(search.path[0]));
-      //} else {
-      let returnCode = creep.moveTo(targetPosObject, {
-        ignoreCreeps: true,
-        costCallback: creep.room.getCostMatrixCallback()
-      });
-      //}
-      return true;
     }
-    creep.say(creep.pos.getDirectionTo(search.path[0]));
-    let returnCode = creep.move(creep.pos.getDirectionTo(search.path[0]));
-  };
+    creep.moveTo(25, 25);
```

Comment:
The center may be obstructed- perhaps a different method is in order.

Reply:
I supose something as creep.moveTo(25, 25, {maxOps: 100}) should avoid to lost the cpu in that case.

Reply:
I usually use `PathFinder` with a range of ~20